File: ENTIDADES/Almacen/ASalidaTransferencia.cs
using ENTIDADES.Generales;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace ENTIDADES.Almacen
{
    [Table("SalidaTransferencia", Schema = "Almacen")]
    public class ASalidaTransferencia: AuditoriaColumna
    {
        [Key]
        public long idsalidatransferencia { get; set; }
        public string codigo { get; set; }
        public DateTime fechatraslado { get; set; }
        public int idsucursal { get; set; }
        public int idalmacensucursalorigen { get; set; }
        public int idsucursaldestino { get; set; }
        public int idempresa { get; set; }
        public int? idcajasucursal { get; set; }
        public int? idcorrelativo { get; set; }
        public string seriedoc { get; set; }
        public string numdoc { get; set; }
        public string estado { get; set; }
        public string estadoguia { get; set; }
        public int? ano { get; set; }
        public string observacion { get; set; }
        public int? idempresatransporte { get; set; }
        public int? idvehiculo { get; set; }
        public int? idtipomovimiento { get; set; }
        public string idcodicentcost { get; set; }
        public decimal? peso_total { get; set; }
        public decimal? bultos { get; set; }


        [ForeignKey("idempresa")]
        public Empresa empresa { get; set; }
        [ForeignKey("idsucursal")]
        public SUCURSAL sucursal { get; set; }
        [NotMapped]
        public EMPLEADO empleado { get; set; }
        [NotMapped]
        public string jsondetalle { get; set; }

    }
}

File: ERP/wwwroot/js/modelos/almacen/ASalidaTransferencia.js

class ASalidaTransferencia {
    constructor() {
        this.idsalidatransferencia = 0;
        this.codigo = '';
        this.fechatraslado = '';
        this.idsucursal = 0;
        this.idalmacensucursalorigen = 0;
        this.idsucursaldestino = 0;
        this.idempresa = 0;
        this.idcaja = 0;
        this.idcorrelativo = 0;
        this.seriedoc = 0;
        this.numdoc = '';
        this.estado = 'HABILITADO';
        this.estadoguia = 'PENDIENTE';
        this.ano = 0;
        this.observacion = '';
        this.idempresatransporte = 0;
        this.idvehiculo = 0;
        this.idcodincentcost=''
    }    
}

class ASalidaTransferenciaDetalle {
    constructor() {

        this.idsalidatransferenciadetalle=0;
        this.idproducto=0;
        this.idstock=0;
        this.cantidad=0;
        this.estado = 'HABILITADO'; ;
        this.idsalidatransferencia=0;
        this.lote='';
        this.regsanitario='';
        this.fechavencimiento='';
        this.idalmacensucursal = 0;
    }   
}